main.nf

```diff
@@ -256,12 +256,39 @@ if (params.deg) {
         .map{ it -> it.sort() }
         .unique()
 }
+
+/*
+* FeatureCounts params
+*/
+if ( params.featurecounts_additional_params.contains('-t ') ) {
+    for (param in params.featurecounts_additional_params.split('-')) {
+        if (param.startsWith('t ')) {
+            gtf_feature_type_ch = Channel.value( param.stripIndent().split(' ')[-1] )
+        }
+    }
+} else {
+    gtf_feature_type_ch = Channel.value('exon')
+}
+if ( params.featurecounts_additional_params.contains('-g ') ){
+    for (param in params.featurecounts_additional_params.split('-')) {
+        if (param.startsWith('g ')){
+            value_of_g = param.stripIndent().split(' ')[-1]
+            gtf_attr_type_ch = Channel.value( value_of_g )
+            gtf_feature_type_of_attr_type_ch = Channel.value( value_of_g.split('_')[0] )
```

Comment:
needed to extract the right entries from the gtf

Reply:
ah okay, so based on the featureCounts params you generate appropriate Channels that are then used in the subsequent annotation preparation processes. cool

groovy syntax, wow :D

Reply:
yeah :relaxed:

but it breaks if you want to sum up to `foobar_id` and there are no `foobar` entries in the annotation :see_no_evil: